.github/workflows/main.yml

```diff
@@ -0,0 +1,52 @@
+name: CI Build
```

Comment:
one last thing, I think it's better to have the name of the yml file match the name of the workflow, so let's rename this from "CI Build" to just "CI", and let's rename the file from "main.yml" to "CI.yml", this way it's easier to add badges to the readme file